src/gui/commandlineparser.cpp

```diff
@@ -131,12 +161,58 @@ CommandLineParser::CommandLineParseResult CommandLineParser::parseCommandLine(Co
         }
     }
 
+    const auto handleLogOption = [this, errorMessage](const QCommandLineOption &option, QtMsgType &result) -> bool {
+        if (d->parser->isSet(option))
+        {
+            const QString logString = d->parser->value(option);
+
+            if (logString == "none")
+            {
+                result = LOG_LEVEL_DISABLED;
+            }
+            else if (logString == "debug")
+            {
+                result = QtDebugMsg;
+            }
+            else if (logString == "info")
+            {
+                result = QtInfoMsg;
+            }
+            else if (logString == "warning")
+            {
+                result = QtWarningMsg;
+            }
+            else if (logString == "critical")
+            {
+                result = QtCriticalMsg;
+            }
+            else if (logString == "fatal")
+            {
+                result = QtFatalMsg;
+            }
+            else
+            {
+                *errorMessage = QObject::tr("Bad log level: ") + logString;
+                return false;
+            }
+        }
+
+        return true;
+    };
+
+    if (!handleLogOption(consoleLogLevelOpion, options->consoleLogLevel)
+        || !handleLogOption(syslogLogLevelOpion, options->syslogLogLevel)
+        || !handleLogOption(fileLogLevelOpion, options->fileLogLevel))
+    {
+        return CommandLineError;
+    }
+
     return CommandLineOk;
 }
 
 void CommandLineParser::showHelp() const
 {
-    d->parser->showHelp();
+    std::cerr << qPrintable(d->parser->helpText());
```

Comment:
Why replace showHelp but not showVersion?

Reply:
`showVersion` does not actually have a version which just returns text. So, I think it will be better to exit in both situations but note that fact in main function